crates/transport-legacy/src/lib.rs

```diff
@@ -2729,6 +2730,137 @@ impl Encode for Value {
     }
 }
 
+// Resource Handles
+#[derive(Debug)]
+pub struct ResourceOwn<T: Send + Sync> {
+    data: Vec<u8>,
+    _ty: PhantomData<T>,
+}
+
+impl<T: Send + Sync> AsRef<[u8]> for ResourceOwn<T> {
+    fn as_ref(&self) -> &[u8] {
+        &self.data
+    }
+}
+
+impl<T: Send + Sync> From<Vec<u8>> for ResourceOwn<T> {
+    fn from(value: Vec<u8>) -> Self {
+        Self {
+            data: value,
+            _ty: PhantomData,
+        }
+    }
+}
+
+impl<T: Send + Sync> From<ResourceOwn<T>> for Vec<u8> {
+    fn from(value: ResourceOwn<T>) -> Self {
+        value.data
+    }
+}
+
+#[async_trait]
+impl<T: Send + Sync> Encode for ResourceOwn<T> {
+    async fn encode(
+        self,
+        payload: &mut (impl bytes::BufMut + Send),
+    ) -> anyhow::Result<Option<AsyncValue>> {
+        self.data.encode(payload).await
+    }
+}
+
+#[async_trait]
+impl<T: Send + Sync> Encode for &ResourceOwn<T> {
+    async fn encode(
+        self,
+        payload: &mut (impl bytes::BufMut + Send),
+    ) -> anyhow::Result<Option<AsyncValue>> {
+        self.data.clone().encode(payload).await
```

Comment:
Probably want to avoid cloning the whole `Vec`, something like this should work?
```suggestion
        self.data.as_slice().encode(payload).await
```